Test/tutorial/parser_combinators.dfy

```diff
@@ -0,0 +1,294 @@
+/// # Parser combinators in Dafny
+///
+/// Challenge: define parse combinators.  As a usability test, use them to parse
+/// expressions of the form `(ⁿ)ⁿ` (`n` opening parentheses followed by `n`
+/// closing parentheses).
+///
+/// This challenge is tricky because we can't write what we'd usually write in
+/// Haskell or OCaml:
+///
```

Comment:
For people who might want to understanding this tutorial and do not yet know about parsing combinators (or need a refresher), we should provide some good references and explain a little more.